compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1610,12 +1612,13 @@ impl<'a> Parser<'a> {
                     e
                 });
                 match field {
-                    Ok(field) => fields.push(field),
+                    Ok(Ok(field)) => fields.push(field),
```

Comment:
`Result<Result<T>>` is, a bit weird imo. I think if possible, we should edit this to represent something similar to `Result<T, Either<DiagnosticBuilder, ErrorGuaranteed>>, and report the error if it's `DiagnosticBuilder`.

Reply:
Let's not do this. I don't even think the eager recovery of fields here is necessarily worth it, given the amount of code that this results in.